### DISCUSSION TITLE
View dependent modelMatrix

### DISCUSSION THREAD

**Discussion by hkfb** [visgl/deck.gl#8675] [Q&A]
I want to be able to scale a layer vertically depending on the view. Eg. I might have two OrbitViews and want to scale a layer differently in those views.

Example:
![image](https://github.com/visgl/deck.gl/assets/19621822/0d61eab1-3da1-4c67-aae8-04af69206aa1)

I can scale using the layer `modelMatrix` but it affects the layer projection in all views. It also affects only one layer, and ideally I would like to scale all layers in a view.

Is there a way to achieve view dependent model scaling?

**Comment by Pessimistress** [comment]
This may be undocumented but I believe you can pass a `modelMatrix` to view state.

**Reply by hkfb** [reply]
I tried this, and it appears to only apply translation. It seems consistent with the code in `viewport.ts` if I read it correctly. Also, in the code it is described as "Model matrix of viewport center".

**Reply by Pessimistress** [reply]
Hmm I think you are correct. You will need to create your own view class to handle this matrix as desired. You can also look at how [OrthographicView](https://github.com/visgl/deck.gl/blob/master/modules/core/src/views/orthographic-view.ts) and [OrthographicViewport](https://github.com/visgl/deck.gl/blob/master/modules/core/src/viewports/orthographic-viewport.ts) implement axis-independent zooms.
